slick/src/main/scala/slick/util/ReadAheadIterator.scala

```diff
@@ -49,3 +43,11 @@ trait ReadAheadIterator[+T] extends BufferedIterator[T] {
     } else throw new NoSuchElementException("next on empty iterator");
   }
 }
+
+object ReadAheadIterator {
```

Comment:
IIRC, implicits in companion objects are always in scope. This means that without explicitly importing everything in this object, users of `ReadAheadIterator` in 2.12 will see two different `headOption` methods that can be called - one from the `BufferedIterator` superclass and one from the `ReadAheadIterator` companion object.

Let me know if that made sense - I could be entirely wrong. Better people than me have been confused by Scala's implicit resolution rules. ;)